elrond-wasm/src/types/interaction/contract_call.rs

```diff
@@ -75,13 +83,25 @@ where
         payment_token: TokenIdentifier,
         payment_amount: SA::AmountType,
     ) -> Self {
-        self.payment_token = payment_token;
-        self.payment_amount = payment_amount;
+        self.payments[0].token_name = payment_token;
+        self.payments[0].amount = payment_amount;
         self
     }
 
     pub fn with_nft_nonce(mut self, payment_nonce: u64) -> Self {
-        self.payment_nonce = payment_nonce;
+        self.payments[0].token_nonce = payment_nonce;
+        self
+    }
+
+    pub fn with_multi_token_transfer(
+        mut self,
+        payments: Vec<EsdtTokenPayment<SA::AmountType>>,
+    ) -> Self {
+        if payments.is_empty() {
+            self.api.signal_error(b"Payment vec should not be empty");
```

Comment:
I'm not sure signaling an error here is the best approach. Maybe just `return self` instead?
```
if !payments.is_empty() {
self.payment = payments;
}

self
```